### PR TITLE
Add ipmi modules for power and boot management

### PR DESCRIPTION
**Pull request by bgaifullin**
- ipmi_power - module for power management
  Parameters:
  - name
  - port
  - user
  - password
  - state
  - timeout
- ipmi_boot - module for boot order management
  Parameters:
  - name
  - port
  - user
  - password
  - bootdev
  - state
  - persist
  - uefi

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @bgaifullin for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by bgaifullin** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @bgaifullin for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: bmc/ipmi/ipmi_boot.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+# ==================================================
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            port=dict(required=False, default=623),
+            user=dict(required=True),
+            password=dict(required=True),
```
Comment (resmo) [inline bmc/ipmi/ipmi_boot.py]: add `no_log=True`for secrets

File: bmc/ipmi/ipmi_power.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            port=dict(required=False, default=623),
+            state=dict(required=False, choices=['on', 'off', 'shutdown', 'reset', 'boot']),
+            user=dict(required=True),
+            password=dict(required=True),
```
Comment (resmo) [inline bmc/ipmi/ipmi_power.py]: add no_log=True for secrets

**Comment by resmo** [comment]
looks pretty solid and straight forward. Minor issue for addig no_log=True for the password arg.

LGTM

File: bmc/ipmi/ipmi_power.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+# ==================================================
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            port=dict(required=False, default=623),
+            state=dict(required=False, choices=['on', 'off', 'shutdown', 'reset', 'boot']),
```
Comment (resmo) [inline bmc/ipmi/ipmi_power.py]: wouldnd it make sense to set required=True for state?

Comment (bgaifullin) [inline bmc/ipmi/ipmi_power.py]: Yes, You are right. Thank you for review. I will do it.

File: bmc/ipmi/ipmi_power.py (hunk 2 above)
Comment (bgaifullin) [inline bmc/ipmi/ipmi_power.py]: I will do. Thank you for review.

**Comment by gregdek** [comment]
Thanks again to @bgaifullin for this PR, and thanks @ansible/core for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: bmc/ipmi/ipmi_boot.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+'''
+
+# ==================================================
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            port=dict(required=False, default=623),
```
Comment (resmo) [inline bmc/ipmi/ipmi_boot.py]: since this is an int, ansible can ensure it is  by adding this: `type='int'`

Comment (bgaifullin) [inline bmc/ipmi/ipmi_boot.py]: done.

**Comment by resmo** [comment]
Thanks!

**Pull request merged**